### PR TITLE
adding lalon masking with prob

### PR DESCRIPTION
**Pull request by cbutsko**
* Adds `Encoder(latlon_mask_prob=0.0)` + a `null_latlon` token; 
* During **training only**, replaces the lat/lon token with the null embedding for `p` of samples to break the location-embedding shortcut;
* Default is 0 (masking off) → no change to existing runs.

### CODE REVIEW

**Comment by uceshaf** [comment]
Two minimal changes done:

- Exposed the latlon_mask_prob parameter so it can be used in finetuning scripts if needed, default is still 0.0
- Added the long pending check of clamping denominator to >=1 so fully-masked timesteps yield a zero embedding rather than NaN (0/0).

Other than that, the modifications seem fine.

**Review by uceshaf — APPROVED** [review]

File: prometheo/models/presto/single_file_presto.py [hunk 1]
```diff
@@ -547,6 +556,18 @@ def forward(
         x, orig_indices, upd_mask = self.mask_tokens(x, mask)
         # append latlon tokens
         latlon_tokens = self.latlon_embed(self.cartesian(latlons)).unsqueeze(1)
+        latlon_mask_prob = getattr(self, "latlon_mask_prob", 0.0)
+        if self.training and latlon_mask_prob > 0.0:
+            # Per-sample null-masking of the location token: with prob
+            # latlon_mask_prob, swap in the learned location-independent
+            # null_latlon so the model cannot rely on coordinates for those
```
Comment (gabrieltseng) [inline prometheo/models/presto/single_file_presto.py:563]: IMO it would be fine to just not have a latlon token here, instead of this learned `null_latlon` token.

**Comment by kvantricht** [comment]
This functionality was implemented in an alternative way in #49 and avoids having to learn additional parameters. Closing this PR.
